### ISSUE TITLE
Adding `fixed` metadata to jscs property

### ISSUE THREAD

**Issue by sindresorhus** [jscs-dev/gulp-jscs#82]
Moved from https://github.com/jscs-dev/gulp-jscs/pull/81#discussion_r28623946

-

I don't really see the point, but @dcneiner can try to convince any of the other maintainers.

**Comment by dcneiner** [comment]
I'll try to circle back this evening or during a break today and add a more succinct explanation for our reasoning on including a `fixed` flag in the jscs medadata. Thanks for moving this into a separate issue and not letting it hold up the primary feature @sindresorhus!

**Comment by dcneiner** [comment]
Ok, I would have titled this issue "Adding `fixed` metadata to `jscs` property":

TL;DR: When using this plugin to fix problems found instead of just reporting them, it would be nice to have a heads up as to which files were actually modified as a result of running jscs on them.

Unlike more normal usages of gulp plugins, a tool like this one (gulp-jscs) that provides a way to fix source files will normally have the `gulp.dest` call result in overwriting the original source files with the fixed versions. This makes using a plugin like `gulp-changed` not super useful since you can’t compare timestamps between a source and destination directory. Other usages might involve passing only changed files in through a gulp watch task or something, but when just running a `format` task or the like, you will often be running against a large number of files, but only changing a few.

While looking at what is handed back from the `fixString` method in `jscs`, it does not seem to expose a way to easily determine if JSCS changed anything in the file. Including a simple comparison of file contents before calling `fixString` and again afterwards allows us to include a small piece of metadata on the existing `jscs` indicating if the file was changed.

The original implementation is here, though I can resubmit this as a PR if there is positive feedback to this issue: https://github.com/LeanKit-Labs/gulp-jscs/blob/8191ed3b21bdd573f1622e2e1f6e0b26ec706b90/index.js#L68-L87

This allows an optional filtering after `jscs` to limit to only the changed files:

```js
var gulp = require('gulp');
var jscs = require('gulp-jscs');
var filter = require('gulp-filter');

gulp.task('default', function () {
    var fixedFilter = filter(function (file) {
        return file.jscs && file.jscs.fixed === true;
    });
    return gulp.src('src/**/*.js')
        .pipe(jscs({
            fix: true
        }))
        .pipe(fixedFilter)
        .pipe(gulp.dest('src'));
});
```

**Comment by shanemikel** [comment]
I would like this behavior.

**Comment by sindresorhus** [comment]
I don't think this is an optimal solution to the actual problem, but if you're still interested, a pull request would be accepted.
